### DISCUSSION TITLE
conversion of units between pixel and cm

### DISCUSSION THREAD

**Discussion by mingjia1025** [jspsych/jsPsych#1006] [Q&A] [answered]
I'm trying to replicate a psychophysical experiment where stimuli were squares of fixed length. It's important that the stimuli I generated by jsPsych have the same physical length regardless of the size of the screen of each subject's devices. To do this, I think I need to know the display size (in cm) and resolution (in pixels) of the subject's screen, so that I can convert between the two units. Is there any function in JsPsych that allows me to access these information?

**Comment by jodeleeuw** [comment, marked as answer]
Sort of.

There's no information in JavaScript that will tell you the display size. You'd need access to information about the monitor, and that's simply not available through JavaScript for security reasons.

I created a plugin that can get this information through another route. It requires the user to hold up an object of known size to the monitor and measure it on the screen. This allows for the conversion to be done. 

[Resize plugin](https://www.jspsych.org/plugins/jspsych-resize/)
